src/components/PreviewWindow/PreviewWindow.tsx

```diff
@@ -46,7 +46,7 @@ const PreviewWindow = () => {
     setTabIndex(newTabIndex)
   }
 
-  const handleOpenDrawer = () => setDrawerOpen(true)
+  const handleOpenDrawer = () => setDrawerOpen(() => !drawerOpen)

```

Comment:
It would be better to rename the handler since it doesn't simply open the drawer now.  
Also, using the previous value inside the callback prevents additional renders when using the state directly.

```suggestion
  const toggleDrawer = () => setDrawerOpen((prev) => !prev)
```

Reply:
Ah, good points! Thanks for the improvements. I've made those updates in a new commit ready for approval.